roadmap/implementers-guide/src/runtime/disputes.md

```diff
@@ -309,15 +302,43 @@ Can be ignored, since all validators are bonded `N` days beyond their duty.
   1. for each `active_head` in `active_heads`
     1. queue a transaction with the `slashing_inherent`
 
+## Partition On-Chain vs Off-Chain
+
+Slashing must happen off-chain, since there is no guarantee that forks of the current selected head, will receive any more
+blocks. As such that block must be scheduled in the transaction pool.
+
+```mermaid
+graph LR;
+    S[Slash]-->T[Transplant slash to all Active Heads]
+```
+
+The slash here is part of a transaction.
+
+```mermaid
+sequenceDiagram
+    participant on as On-Chain
+    participant off as Off-Chain
+    on ->> off: offchain_index::set(b"k", b"v")
+
+    activate off
+    off ->> on: trait OffchainSubmitTransaction::submit_at(..)
+    Note over on,off: with args:<br> `at: Block` current<br>`source: TransactionSource::Local`<br>`xts: <_>::Extrinsic::IncludeData(<encoded data for On-Chain>)`
+    deactivate off
+```
+
+Transitioning the gap from on-chain to off-chain is done by two means.
+The offchain indexing API to move data from On-Chain to deferred Off-Chain
+consumption.
+This has to be done for the validator sets, plus the data of interest (TODO what's that exactly?) that is going to be part of the block and record the meshbehaviour of the minority of the quorums as defined earlier.
+
+In the reverse, from off-chain to on-chain, crafting a direct transaction achieves the information crossing by passing a scale encoded structure containing sufficient data to identify the offenders.
 
-# Open Questions
+## Secondary Approval Checking
 
-  - what happens with open disputes on session / era change?
-  - > TODO: validator-dispute could be instead replaced by a fisherman w/ bond
-  - > TODO: Given that a remote dispute is likely to be replayed across multiple forks, it is important to choose a VRF in a way that all forks processing the remote dispute will have the same one. Choosing the VRF is important as it should not allow an adversary to have control over who will be selected as a secondary approval checker.
-  - decided and describe **era** change validator set change resolution
-  - decided and describe **session** change validator set change resolution
-  - who can dispute, validators of the current session? Or also validators of the previous session?
+Given that a remote dispute is likely to be replayed across multiple forks, it is important to choose a VRF in a way that all forks processing the remote dispute will have the same output of the VRF process.
+Choosing the VRF is important as it should not allow an adversary to have control over who will be selected as a secondary approval checker.
+
+The VRF is defined by / part of the consensus algorithm.
```

Comment:
Appears our VRF comments got promoted from TODOs to text, when they should probably have been removed, or moved to an approvals related file.

Reply:
Going to drop.